transport_test.go

```diff
@@ -42,3 +57,143 @@ func TestInitTransportWithCustomCA(t *testing.T) {
 		t.Error("expected root CA to be set, got nil")
 	}
 }
+
+type testHTTPHandler struct {
+	t                 *testing.T
+	gotCorrectRequest chan (bool)
+}
+
+func (h *testHTTPHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	if len(req.TLS.PeerCertificates) > 0 {
+		h.gotCorrectRequest <- true
+	} else {
+		reqDump, _ := httputil.DumpRequest(req, false)
+		h.t.Logf("got request without client certificates:\n%s", reqDump)
+		h.t.Logf("TLS config: %#v", req.TLS)
+
+		h.gotCorrectRequest <- false
+	}
+}
+
+func TestInitTransportWithClientCertAuth(t *testing.T) {
```

Comment:
I think it is amazing to have a test like that, but it makes me wonder if we are testing our code or the golang libraries.

Reply:
I got something very similar wrong in the past, and it was an issue with the code.